From: Felix Riemann <friemann@gnome.org>
Date: Sat, 6 Apr 2019 02:13:00 +0200
Subject: EogWindow: Don't unref timer sources before destroying them

This could cause segfaults if the source was no longer attached
to the mainloop. Fixes #50.
---
 src/eog-window.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/eog-window.c b/src/eog-window.c
index c066556..775eb72 100644
--- a/src/eog-window.c
+++ b/src/eog-window.c
@@ -1791,8 +1791,8 @@ fullscreen_clear_timeout (EogWindow *window)
 	eog_debug (DEBUG_WINDOW);
 
 	if (window->priv->fullscreen_timeout_source != NULL) {
-		g_source_unref (window->priv->fullscreen_timeout_source);
 		g_source_destroy (window->priv->fullscreen_timeout_source);
+		g_source_unref (window->priv->fullscreen_timeout_source);
 	}
 
 	window->priv->fullscreen_timeout_source = NULL;
@@ -1823,8 +1823,8 @@ slideshow_clear_timeout (EogWindow *window)
 	eog_debug (DEBUG_WINDOW);
 
 	if (window->priv->slideshow_switch_source != NULL) {
-		g_source_unref (window->priv->slideshow_switch_source);
 		g_source_destroy (window->priv->slideshow_switch_source);
+		g_source_unref (window->priv->slideshow_switch_source);
 	}
 
 	window->priv->slideshow_switch_source = NULL;
